### PR TITLE
Feature: Add edit popup

### PR DESCRIPTION
**Pull request by MrF3lix**
- Allow to edit name 
- Allow to delete data flow
- Fix onChange handlers
- Add component stories    

## Definition of Done

- [x] User Story
  - [x] All Acceptance Criteria met
- [x] Code
  - [x] No more code needed
  - [x] No known bugs
- [x] Clean Code
  - [x] Comments have been created for new code
  - [x] ESLinter found no issues (editor & executor only)
  - [x] No unavoidable Code Smells
- [x] Testing
  - [x] Existing Tests pass
  - [x] New Unit Tests created
  - [x] New Unit Tests pass
- [x] SonarCloud Quality Gate passed

## Definition of more than Done (optional)

- [ ] Testing
  - [ ] New Integration Tests created
  - [ ] New Integration Tests passed

### CODE REVIEW

File: src/editor/pages/api/flows/delete.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { NextApiRequest, NextApiResponse } from 'next'
+
+const handler = async ( req: NextApiRequest, res: NextApiResponse ): Promise<void> => {
+  const response = await fetch( `${process.env.API_BASE_URL}/api/DataFlow/${req.body.id}`, {
+    method: req.method,
```
Comment (nikipuk) [inline src/editor/pages/api/flows/delete.ts:5]: I'd say the delete call should always be "DELETE"

File: src/editor/components/list/list-item-status.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+}
+
+export const ListItemStatus = ( { status }: ListItemStatusProps ): JSX.Element => {
+  const getStatusIcon = ( ): JSX.Element => {
+    switch ( status ) {
+      case 'Active':
+        return <CheckCircleIcon className="text-emerald-400 h-4 w-4" />
+      case 'Disabled':
+        return <XCircleIcon className="text-red-400 h-4 w-4" />
+    }
```
Comment (nikipuk) [inline src/editor/components/list/list-item-status.tsx:14]: should we do status.lower() and check agauinst 'acctive' and 'disabled' to remove case sensitivity?

File: src/editor/components/flow/edit-dialog.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        <Input
+          label="Repeat the Data Flow name to delete it."
+          value={deleteFlowName}
+          placeholder={flow.name}
+          onChange={( e ): void => setDeleteFlowName( e.target.value )}
+        />
+        <Button dangerous disabled={deleteFlowName !== flow.name} small onClick={deleteFlow}>
+          Delete Data Flow
+        </Button>
+      </div>
```
Comment (nikipuk) [inline src/editor/components/flow/edit-dialog.tsx:56]: this div is missing a red outline 😢

File: src/editor/components/flow/edit-dialog.tsx [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+  save(): void
+  flow: any
+  setFlow( value: any ): void
+  deleteFlow(): void
+}
+
+const states = [
+  { value: 'Active', name: 'Active' },
+  { value: 'Disabled', name: 'Disabled' }
+]
```
Comment (nikipuk) [inline src/editor/components/flow/edit-dialog.tsx:19]: maybe this should be defined somewhere else so other components using the types uf states can access it

File: src/editor/components/list/list-item-status.tsx (hunk 2 above)
Comment (MrF3lix) [reply src/editor/components/list/list-item-status.tsx:14]: For now I think it's fine to leave it like this. Later we will use an ENUM for this anyway.

File: src/editor/components/flow/edit-dialog.tsx (hunk 4 above)
Comment (MrF3lix) [reply src/editor/components/flow/edit-dialog.tsx:19]: Yes, this should be an ENUM at some point.

File: src/editor/components/list/list-item-status.tsx (hunk 2 above)
Comment (nikipuk) [reply src/editor/components/list/list-item-status.tsx:14]: ok, we'll do this in a refactor / feature

**Review by nikipuk — APPROVED** [review]
looks good

**Pull request merged**